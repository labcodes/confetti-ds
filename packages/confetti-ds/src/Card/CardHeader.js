import React from "react";
import PropTypes from "prop-types";

import Icon from "../Icon";
import { SimpleTag } from "../Tags";
import { ICON_TYPES } from "../constants";

export default class CardHeader extends React.Component {
  static propTypes = {
    /** Sets card's title. */
    title: PropTypes.string,
    /** Sets a className for card's title. */
    titleClassName: PropTypes.string,
    /** Sets card's subtitle. */
    subtitle: PropTypes.string,
    /** Sets a className for card's subtitle. */
    subtitleClassName: PropTypes.string,
    /** Sets card's category. */
    categoryText: PropTypes.string,
    /** Sets card's label. */
    categoryLabelText: PropTypes.string,
    /** Sets card's category icon. */
    categoryIcon: PropTypes.oneOf(ICON_TYPES),
    /** Sets card's category badge color. */
    categoryColor: PropTypes.oneOf(["mineral", "teal", "purple"]),
    /** Sets whether the header will go over the image or not. */
    isOverlay: PropTypes.bool,
  };

  static defaultProps = {
    title: undefined,
    titleClassName: undefined,
    subtitle: undefined,
    subtitleClassName: undefined,
    categoryText: undefined,
    categoryLabelText: undefined,
    categoryIcon: undefined,
    categoryColor: undefined,
    isOverlay: undefined,
  };

  constructor(props) {
    super(props);
    this.checkColorAndIcon(props);
  }

  componentDidUpdate() {
    this.checkColorAndIcon(this.props);
  }

  checkColorAndIcon = (props) => {
    const { categoryColor, categoryIcon } = props;
    if (categoryIcon && categoryColor) {
      throw Error(
        "A CardHeader can't have both `categoryColor` and `categoryIcon`. Please choose one of them."
      );
    }
  };

  render() {
    const {
      title,
      titleClassName,
      subtitle,
      subtitleClassName,
      categoryText,
      categoryLabelText,
      categoryIcon,
      categoryColor,
      isOverlay,
    } = this.props;

    return (
      <article
        className={`lab-card-header${
          isOverlay ? " lab-card-header--overlay" : ""
        }`}
      >
        {categoryIcon || categoryColor || categoryText || categoryLabelText ? (
          <p className="lab-card-category">
            {categoryIcon ? <Icon type={categoryIcon} /> : null}
            {categoryColor ? (
              <span
                className={`lab-card-category__color lab-card-category__color--${categoryColor}`}
              />
            ) : null}
            {categoryText ? (
              <span className="lab-card-category__text">{categoryText}</span>
            ) : null}
            {categoryLabelText ? <SimpleTag text={categoryLabelText} /> : null}
          </p>
        ) : null}
        {title ? (
          <p
            className={`lab-card-title${
              titleClassName ? ` ${titleClassName}` : ""
            }`}
          >
            {title}
          </p>
        ) : null}
        {subtitle ? (
          <p
            className={`lab-card-subtitle${
              subtitleClassName ? ` ${subtitleClassName}` : ""
            }`}
          >
            {subtitle}
          </p>
        ) : null}
      </article>
    );
  }
}
