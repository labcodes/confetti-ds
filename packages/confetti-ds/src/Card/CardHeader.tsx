import React, { useEffect } from "react";

import Icon from "../Icon";
import { SimpleTag } from "../Tags";
import { IconTypes } from "../constants";

interface CardHeaderProps {
  /** Sets card's title. */
  title?: string;
  /** Sets a className for card's title. */
  titleClassName?: string;
  /** Sets card's subtitle. */
  subtitle?: string;
  /** Sets a className for card's subtitle. */
  subtitleClassName?: string;
  /** Adds a category text in the Card Header. Used to define the group of which the Card is part of. May use a categoryIcon OR categoryColor to the left as visual support for its content. */
  categoryText?: string;
  /** Sets a SimpleTag's text in the Card Header. Can be used instead of categoryText to highlight the group of which the Card is part of OR together with categoryText as a sub-category. Only use if categoryColor is not present.  */
  categoryTagText?: string;
  /** Always used with categoryTagText to set SimpleTag's color in the Card Header. If unset the Tag will be displayed with its default color. */
  categoryTagColor?: string;
  /** Defines an Icon as visual support to categoryText in the Card Header. It must not be used together with categoryColor. */
  categoryIcon?: IconTypes;
  /** Defines a Color as visual support to categoryText in the Card Header. It must not be used together with categoryIcon. */
  categoryColor?: string;
  /** Sets whether the header will go over the image or not. */
  isOverlay?: boolean;
}

export default function CardHeader({
  title,
  titleClassName,
  subtitle,
  subtitleClassName,
  categoryText,
  categoryTagText,
  categoryTagColor,
  categoryIcon,
  categoryColor,
  isOverlay,
}: CardHeaderProps) {
  const checkColorAndIcon = (categoryColor, categoryIcon) => {
    if (categoryIcon && categoryColor) {
      throw Error(
        "A CardHeader can't have both `categoryColor` and `categoryIcon`. Please choose one of them."
      );
    }
  };

  useEffect(() => {
    checkColorAndIcon(categoryColor, categoryIcon);
  }, [categoryColor, categoryIcon]);

  return (
    <article
      className={`lab-card-header${
        isOverlay ? " lab-card-header--overlay" : ""
      }`}
    >
      {categoryIcon || categoryColor || categoryText || categoryTagText ? (
        <p className="lab-card-category">
          {categoryIcon ? <Icon type={categoryIcon} /> : null}
          {categoryColor ? (
            <span
              className={`lab-card-category__color lab-card-category__color--${categoryColor}`}
            />
          ) : null}
          {categoryText ? (
            <span className="lab-card-category__text">{categoryText}</span>
          ) : null}
          {categoryTagText ? (
            <SimpleTag text={categoryTagText} color={categoryTagColor} />
          ) : null}
        </p>
      ) : null}
      {title ? (
        <p
          className={`lab-card-title${
            titleClassName ? ` ${titleClassName}` : ""
          }`}
        >
          {title}
        </p>
      ) : null}
      {subtitle ? (
        <p
          className={`lab-card-subtitle${
            subtitleClassName ? ` ${subtitleClassName}` : ""
          }`}
        >
          {subtitle}
        </p>
      ) : null}
    </article>
  );
}
