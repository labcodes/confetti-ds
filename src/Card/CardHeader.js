import React, { useEffect } from "react";
import PropTypes from "prop-types";

import Icon from "../Icon";
import { SimpleTag } from "../Tags";
import { ICON_TYPES } from "../constants";

export default function CardHeader({
  title,
  titleClassName,
  subtitle,
  subtitleClassName,
  categoryText,
  categoryTagText,
  categoryTagColor,
  categoryIcon,
  categoryColor,
  isOverlay,
}) {
  const checkColorAndIcon = (categoryColor, categoryIcon) => {
    if (categoryIcon && categoryColor) {
      throw Error(
        "A CardHeader can't have both `categoryColor` and `categoryIcon`. Please choose one of them."
      );
    }
  };

  useEffect(() => {
    checkColorAndIcon(categoryColor, categoryIcon);
  }, [categoryColor, categoryIcon]);

  return (
    <article
      className={`lab-card-header${
        isOverlay ? " lab-card-header--overlay" : ""
      }`}
    >
      {categoryIcon || categoryColor || categoryText || categoryTagText ? (
        <p className="lab-card-category">
          {categoryIcon ? <Icon type={categoryIcon} /> : null}
          {categoryColor ? (
            <span
              className={`lab-card-category__color lab-card-category__color--${categoryColor}`}
            />
          ) : null}
          {categoryText ? (
            <span className="lab-card-category__text">{categoryText}</span>
          ) : null}
          {categoryTagText ? (
            <SimpleTag text={categoryTagText} color={categoryTagColor} />
          ) : null}
        </p>
      ) : null}
      {title ? (
        <p
          className={`lab-card-title${
            titleClassName ? ` ${titleClassName}` : ""
          }`}
        >
          {title}
        </p>
      ) : null}
      {subtitle ? (
        <p
          className={`lab-card-subtitle${
            subtitleClassName ? ` ${subtitleClassName}` : ""
          }`}
        >
          {subtitle}
        </p>
      ) : null}
    </article>
  );
}

CardHeader.propTypes = {
  /** Sets card's title. */
  title: PropTypes.string,
  /** Sets a className for card's title. */
  titleClassName: PropTypes.string,
  /** Sets card's subtitle. */
  subtitle: PropTypes.string,
  /** Sets a className for card's subtitle. */
  subtitleClassName: PropTypes.string,
  /** Adds a category text in the Card Header. Used to define the group of which the Card is part of. May use a categoryIcon OR categoryColor to the left as visual support for its content. */
  categoryText: PropTypes.string,
  /** Sets a SimpleTag's text in the Card Header. Can be used instead of categoryText to highlight the group of which the Card is part of OR together with categoryText as a sub-category. Only use if categoryColor is not present.  */
  categoryTagText: PropTypes.string,
  /** Always used with categoryTagText to set SimpleTag's color in the Card Header. If unset the Tag will be displayed with its default color. */
  categoryTagColor: PropTypes.string,
  /** Defines an Icon as visual support to categoryText in the Card Header. It must not be used together with categoryColor. */
  categoryIcon: PropTypes.oneOf(ICON_TYPES),
  /** Defines a Color as visual support to categoryText in the Card Header. It must not be used together with categoryIcon. */
  categoryColor: PropTypes.string,
  /** Sets whether the header will go over the image or not. */
  isOverlay: PropTypes.bool,
};

CardHeader.defaultProps = {
  title: undefined,
  titleClassName: undefined,
  subtitle: undefined,
  subtitleClassName: undefined,
  categoryText: undefined,
  categoryTagText: undefined,
  categoryTagColor: undefined,
  categoryIcon: undefined,
  categoryColor: undefined,
  isOverlay: undefined,
};
