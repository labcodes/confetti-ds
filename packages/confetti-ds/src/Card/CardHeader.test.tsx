import React from "react";
import renderer from "react-test-renderer";

import CardHeader from "./CardHeader";

describe("CardHeader", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer.create(<CardHeader />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with title", async () => {
    const renderedComponent = renderer
      .create(<CardHeader title="test" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with titleClassName", async () => {
    const renderedComponent = renderer
      .create(<CardHeader titleClassName="test-class-name" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with subtitle", async () => {
    const renderedComponent = renderer
      .create(<CardHeader subtitle="test" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with subtitleClassName", async () => {
    const renderedComponent = renderer
      .create(<CardHeader subtitleClassName="test-class-name" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with categoryText", async () => {
    const renderedComponent = renderer
      .create(<CardHeader categoryText="test" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with categoryTagText", async () => {
    const renderedComponent = renderer
      .create(<CardHeader categoryTagText="test" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with categoryIcon", async () => {
    const renderedComponent = renderer
      .create(<CardHeader categoryIcon="star" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with categoryColor", async () => {
    const renderedComponent = renderer
      .create(<CardHeader categoryColor="mineral" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isOverlay", async () => {
    const renderedComponent = renderer
      .create(<CardHeader isOverlay />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
