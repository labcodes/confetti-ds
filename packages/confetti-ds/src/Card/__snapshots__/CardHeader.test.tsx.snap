// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader renders with base props 1`] = `
<article
  className="lab-card-header"
/>
`;

exports[`CardHeader renders with categoryColor 1`] = `
<article
  className="lab-card-header"
>
  <p
    className="lab-card-category"
  >
    <span
      className="lab-card-category__color lab-card-category__color--mineral"
    />
  </p>
</article>
`;

exports[`CardHeader renders with categoryIcon 1`] = `
<article
  className="lab-card-header"
>
  <p
    className="lab-card-category"
  >
    <span
      className="lab-icon lab-icon--star  "
    >
      <svg
        aria-hidden={true}
        className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height={20}
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 2a8 8 0 1 1 0 16 8 8 0 0 1 0-16Zm0 1a7 7 0 1 0 0 14 7 7 0 0 0 0-14Zm0 9.5a.75.75 0 1 1 0 1.5.75.75 0 0 1 0-1.5ZM10 6a.5.5 0 0 1 .5.41V11a.5.5 0 0 1-1 .09V6.5c0-.28.22-.5.5-.5Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </p>
</article>
`;

exports[`CardHeader renders with categoryTagText 1`] = `
<article
  className="lab-card-header"
>
  <p
    className="lab-card-category"
  >
    <span
      className="lab-tag  lab-tag--pale"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      test
    </span>
  </p>
</article>
`;

exports[`CardHeader renders with categoryText 1`] = `
<article
  className="lab-card-header"
>
  <p
    className="lab-card-category"
  >
    <span
      className="lab-card-category__text"
    >
      test
    </span>
  </p>
</article>
`;

exports[`CardHeader renders with isOverlay 1`] = `
<article
  className="lab-card-header lab-card-header--overlay"
/>
`;

exports[`CardHeader renders with subtitle 1`] = `
<article
  className="lab-card-header"
>
  <p
    className="lab-card-subtitle"
  >
    test
  </p>
</article>
`;

exports[`CardHeader renders with subtitleClassName 1`] = `
<article
  className="lab-card-header"
/>
`;

exports[`CardHeader renders with title 1`] = `
<article
  className="lab-card-header"
>
  <p
    className="lab-card-title"
  >
    test
  </p>
</article>
`;

exports[`CardHeader renders with titleClassName 1`] = `
<article
  className="lab-card-header"
/>
`;
